types/time.go

```diff
@@ -596,9 +635,347 @@ func splitDateTime(format string) (seps []string, fracStr string) {
+				var currentPart = format[start:i]
+				currentSepChar = format[i]
+
+				if sepsLen == 0 {
+					if '-' == lastSepChar || '-' == currentSepChar {
+						seps = append(seps, currentPart)
+						totalPartCount++
+					} else {
+
+						var partsWithYear, validLen, err = ParseNoDelimiterHavingYear(currentPart)
+						if err != nil {
+							return nil, realFsp, err
+						}
+
+						for j := 0; j < validLen; j++ {
+							seps = append(seps, partsWithYear[j])
+						}
+						totalPartCount = validLen
+					}
+				} else {
+					if sepsLen == 6 {
+						if '.' == lastSepChar {
+							if selfDecideFsp {
+								if '0' == currentPart[0] {
+									realFsp = len(currentPart)
+									if realFsp > MaxFsp {
+										realFsp = MaxFsp
+									}
+								} else {
+									realFsp = MaxFsp
+								}
+							}
+							seps = append(seps, currentPart)
+						}
+						return seps, realFsp, nil
+					} else {
+						if '-' == lastSepChar || '-' == currentSepChar {
+							seps = append(seps, currentPart)
+							totalPartCount++
+						} else {
+							var partsNoYear, validLen, err = ParseNoDelimiterNoYear(currentPart)
+							if err != nil {
+								return nil, realFsp, err
+							}
+							for j := 0; j < maxPartCount-sepsLen && j < validLen; j++ {
+								seps = append(seps, partsNoYear[j])
+								totalPartCount++
+							}
+						}
+					}
+				}
+
+				if totalPartCount >= maxPartCount {
+					return seps, realFsp, nil
+				}
+			}
+		}
+	}
+
+	var currentPart = format[start:]
+	var sepsLen = len(seps)
+	if sepsLen == 0 {
+		var partsWithYear, validLen, err = ParseNoDelimiterHavingYear(currentPart)
+		if err != nil {
+			return nil, realFsp, err
+		}
+
+		for j := 0; j < validLen; j++ {
+			seps = append(seps, partsWithYear[j])
+		}
+		totalPartCount = validLen
+	} else if 6 == sepsLen {
+		if '.' == lastSepChar {
+			if selfDecideFsp {
+				if '0' == currentPart[0] {
+					realFsp = len(currentPart)
+					if realFsp > MaxFsp {
+						realFsp = MaxFsp
+					}
+				} else {
+					realFsp = MaxFsp
+				}
+			}
+
+			seps = append(seps, currentPart)
+		}
+
+		return seps, realFsp, nil
+	} else {
+		var partsNoYear, validLen, err = ParseNoDelimiterNoYear(currentPart)
+		if err != nil {
+			return nil, realFsp, err
+		}
+		for j := 0; j < maxPartCount-sepsLen && j < validLen; j++ {
+			seps = append(seps, partsNoYear[j])
+			totalPartCount++
+		}
+	}
+
+	return seps, realFsp, nil
+}
+
+func ParseNoDelimiterNoYear(strNoDelimiter string) ([6]string, int, error) {
+	var seps = [6]string{"", "", "", "", "", ""}
+	var validLen = 0 // valid length
+	var err error
+	var month = &seps[0]
+	var day = &seps[1]
+	var hour = &seps[2]
+	var minute = &seps[3]
+	var second = &seps[4]
+	var microsecond = &seps[5]
+	len := len(strNoDelimiter)
+
+	var micsFormat = "%" + strconv.Itoa(MaxFsp) + "s"
+
+	if len > 11 {
+		_, err = fmt.Sscanf(strNoDelimiter, "%2s%2s%2s%2s%2s"+micsFormat, month, day, hour, minute, second, microsecond)
```

Comment:
`seps` array is not necessary, how about that:
```
var year, month, day, hour, minute, second string
fmt.Sscanf(strNoDelimiter, "%2s%2s%2s%2s%2s"+micsFormat, &month, &day, &hour, &minute, &second, &microsecond)
```

Reply:
not very good.Because ParseNoDelimiterNoYear need to return a string array.